### PR TITLE
Prefer nullish coalescing and optional chaining

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Enforce nullish coalescing and optional chaining linting rules.

## Why?

They are the default in the `@guardian/eslint-config-typescript` and convey the intent of the logic better.

Part of #4935 and #5863

### CODE REVIEW

File: dotcom-rendering/src/web/components/Callout/Select.tsx [hunk 1]
```diff
@@ -25,12 +25,11 @@ export const Select = ({ formField, formData, setFormData }: Props) => (
...
-				formField.options.map((option, index) => (
-					<option key={index} value={option.value}>
-						{option.value}
-					</option>
-				))}
+			{formField.options.map((option, index) => (
+				<option key={index} value={option.value}>
+					{option.value}
+				</option>
+			))}
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Callout/Select.tsx:32]: The compiler thinks this is always defined

File: dotcom-rendering/src/web/components/ReaderRevenueDev.importable.tsx [hunk 2]
```diff
@@ -28,7 +28,8 @@ export const ReaderRevenueDev = ({ shouldHideReaderRevenue }: Props) => {
 						),
 					);
 
-		if (window && window.guardian) {
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- window.guardian may be missing?
+		if (window?.guardian) {
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/ReaderRevenueDev.importable.tsx:32]: Not sure about that one, but I assumed it was added for a reason

File: dotcom-rendering/src/web/lib/stringifyFileBase64.ts [hunk 3]
```diff
@@ -4,10 +4,9 @@ export const stringifyFileBase64 = (file: File): Promise<string> =>
 		reader.addEventListener(
 			'load',
 			() => {
-				const fileAsBase64 =
-					reader &&
-					reader.result &&
-					reader.result.toString().split(';base64,')[1];
+				const fileAsBase64 = reader.result
+					?.toString()
+					.split(';base64,')[1];
```
Comment (mxdvl) [inline dotcom-rendering/src/web/lib/stringifyFileBase64.ts:9]: The compiler thinks reader is always defined

File: dotcom-rendering/src/model/find-pillar.ts [hunk 4]
```diff
@@ -6,10 +6,12 @@ export const findPillar: (
 	tags?: TagType[],
 ) => ArticleTheme | undefined = (name, tags?) => {
 	// Flag paid content for Labs pillar (for styling purposes)
-	const isPaidContent = (tag: TagType) =>
-		tag.type === 'Tone' && tag.id === 'tone/advertisement-features';
+	const hasPaidContent = !!tags?.some(
```
Comment (sndrs) [inline dotcom-rendering/src/model/find-pillar.ts:9]: doesn't this coerce `tags` into a boolean? can you call `some` on that?

Comment (mxdvl) [reply dotcom-rendering/src/model/find-pillar.ts:9]: [TypeScript thinks it’s fine](https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCGBzCMC8MCy8oAsB0ATvGACYgC2AFAJQwB8MADHgKwwD8MA2gN4wCWJAFwwARFHABTAPTwSAN0kEo-CJPKSwUUTAC+AXRgiArqUkAzfmEkkAULdCQQAG0l5nIRJQCE3hMnY8CApJSko+QT1aVAZI1HixCWtZBSUVNQ0tUWpqIA)

File: dotcom-rendering/.eslintrc.js [hunk 5]
```diff
@@ -23,13 +23,8 @@ const rulesToOverrideGuardianConfig = {
 		},
 	],
 
+	// This is not safe to remove whilst we have noUncheckedIndexedAccess
 	'@typescript-eslint/no-unnecessary-condition': 'warn',
```
Comment (OllysCoding) [inline dotcom-rendering/.eslintrc.js:27]: Can this change now then?

**Review by OllysCoding — APPROVED** [review]
LGTM :) Nice tidy!

**Review by AshCorr — APPROVED** [review]
Love it!

File: dotcom-rendering/.eslintrc.js (hunk 5 above)
Comment (mxdvl) [reply dotcom-rendering/.eslintrc.js:27]: Waiting on your PR 😉

**Pull request merged**